tree_hugger/core/parser/java/java_parser.py

```diff
@@ -60,31 +67,38 @@ def get_all_method_names_with_params(self) -> Dict[str, str]:
         """
         captures = self._run_query_and_get_captures('all_method_names_and_params', self.root_node)
         ret_struct = {}
-        for i in range(0, len(captures), 2):
-            method_name = match_from_span(captures[i][0], self.splitted_code)
-            ret_struct[method_name] = []
-            for param in captures[i+1][0].children:
-                if param.type == "formal_parameter":
-                    name = match_from_span(
-                        param.child_by_field_name("name"),
-                        self.splitted_code
-                    )
-                    typ = match_from_span(
-                        param.child_by_field_name("type"),
-                        self.splitted_code
-                    )
-                elif param.type == "spread_parameter":
-                    name = match_from_span(
-                        param.children[2],
-                        self.splitted_code
-                    )
-                    typ = match_from_span(
-                        param.children[0],
-                        self.splitted_code
-                    )
-                else:
-                    continue
-                ret_struct[method_name].append((name,typ))
+        i = 0
+        while i < len(captures):
```

Comment:
Same here, not sure if it is totally possible. But if you can figure it out then it will read more Pythonic